### ISSUE TITLE
P2P (both EV and batteries), P2H and P2G should only store/convert excess

### ISSUE THREAD

**Issue by ChaelKruip** [quintel/etloader#121]
This is not really a question but more a request for verification. :smile: 

What I mean by excess electricity is the production of solar PV (or other volatile producers) that cannot be used by consuming technologies and would otherwise be sent to the network. So, say at time 't' we have a connection of the network with
* P2P battery
* solar PV producing 10 kW
* Television consuming 10 kW
In this case, the P2P battery should not be storing anything.

If the situation is as follows:
* P2P battery
* solar PV producing 15 kW
* Television consuming 10 kW
the battery should store at a rate of 5 kW (if its capacity permits). For realistic batteries, the capacity is about 2 kW however so it would only store at that rate and 3 kW load would flow to the network.

@antw is this indeed how it works?

**Comment by antw** [comment]
> @antw is this indeed how it works?

TL;DR: Yes!

There are two types of consumption: "mandatory" and "conditional".

* **Mandatory consumption** is that which *must* be satisfied; if there is insufficient production in the testing ground, the grid will be used to balance the deficit. If you have a television consuming 10 kW and a solar PV producing 5 kW, 5kW will come from the PV and 5 kW from the grid.

* **Conditional consumption** is that which is satisfied *only if there is an excess of production within the testing ground*; never using the grid. Batteries, P2H, and P2G are all classed as "conditional". If there is no excess, they will receive nothing.

Electric vehicles have both mandatory *and* conditional consumption. The load required to satisfy the minimum charge of the vehicle (defined by the availability profile) is *mandatory*. If there is not enough production in the testing ground for the car to charge to its minimum stored volume, it will use the grid. The car will only charge *beyond* its minimum volume if there is an excess of production.

To put it another way: the area below the availability curve is mandatory and *must* be fulfilled. The area above the curve is conditional.

![ev-load](https://cloud.githubusercontent.com/assets/4383/7655110/69ffe376-fb19-11e4-89c9-91fb3e54658a.jpg)

Let me know if this is unclear. :smiley:

**Comment by antw** [comment]
Oh... that red line should read "Volume", not "Capacity". /faceplam

**Comment by ChaelKruip** [comment]
> Let me know if this is unclear. :smiley:

Very clear thanks!

**Comment by ChaelKruip** [comment]
Closing.
